fix: add org_id to the options

avisiedo opened the pull request:
When we register a domain, we need to specify the org_id for the generated identity; the org_id we use here must match the org_id used when generating the token or the operation will fail with sign mismatching.

This change add a second option to specify the ORG_ID to be used for the identity.

avisiedo committed:
refactor: fixes and improves maintenance

When we register a domain, we need to specify the org_id for the generated
identity; the org_id we use here must match the org_id used when generating the
token or the operation will fail with sign mismatching; now a ORG_ID environment
variable is used, that by default will be 12345 for better experience, but its
value can be override from the CLI.

This change additionally remove duplicated code and keep the scripts cleaner.
Bear in mind that 'curl.sh' wrapper prepare the request depending on the
environment variables if moving additional variables to the common scripts.

- Allows to customize ORG_ID variable; it could be helpful when checking data
  isolation.
- Allows to customize X_RH_IDENTITY and X_RH_FAKE_IDENTITY; it could be helpful
  when checking identity enforcement.
- Now the scripts can be invoked from anywhere in the repository.

Co-authored-by: Christian Heimes <cheimes@redhat.com>
Signed-off-by: Alejandro Visiedo <avisiedo@redhat.com>

tiran reviewed (changes requested):
- All `local-*` should use the same org id. The org id for local testing should be consistent. You could replace the hard-coded string with "${ORG_ID:-12345}" to allow overrides with an env var.
- If the org_id is important for ephemeral, than all `ephe-*` scripts must take an org id.

tiran commented:
See PR #8

avisiedo commented:
thanks mate for the review. Some conversations with @aadhikar  made me think about some wrong things into the scripts (many thanks).

@tiran I have taken a few things from your MR, I close that because this add a deeper refactor and enhance the maintenance in the future, removing some duplication and make the customization better between all the different scripts, but I appreciate a lot your suggestions :+1: (they are included here).

I close: https://github.com/podengo-project/idmsvc-backend/pull/8

tiran commented on test/scripts/common.inc.sh:
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
```
Shell include files should neither have a shebang nor should they have executable permissions. I would also name the files `.inc` instead of `.inc.sh` to indicate that they are include files.

avisiedo replied:
Thanks mate! I remove the shebang and rename them; about executable permissions, they include files does not have executable permissions.

tiran commented on test/scripts/common.inc.sh:
```diff
@@ -0,0 +1,17 @@
+
+ORG_ID="${ORG_ID:-12345}"
+
+IDENTITY_USER="$( ./tools/bin/xrhidgen -org-id "${ORG_ID}" user -is-active=true -is-org-admin=true -user-id test -username test | base64 -w0 )"
```
I don't like the fact that you call the tool unconditionally. You can do

```
user_identity() {
    ${XRHIDGEN} -org-id "${ORG_ID}" user -is-active=true -is-org-admin=true -user-id test -username test | base64 -w0
}
```

and then assign the output on demand with

```
X_RH_IDENTITY=$(user_identity)
```

This works because the command outputs to stdout and `$()` returns stdout for the assignment operator.

avisiedo replied:
JFTR: I had to export the function to make it available in the subshell

tiran commented on test/scripts/common.inc.sh:
```diff
@@ -0,0 +1,17 @@
+ORG_ID="${ORG_ID:-12345}"
+
+IDENTITY_USER="$( ./tools/bin/xrhidgen -org-id "${ORG_ID}" user -is-active=true -is-org-admin=true -user-id test -username test | base64 -w0 )"
+[ "${IDENTITY_USER}" != "" ] || error "IDENTITY_USER got empty: check that you executed 'make install-tools'"
```
And pipes strike again! All scripts should set `set -eo pipefail` (exit on uncatched error, catch pipe failures).

In this case, it would be better to add an explicit check for presence of the binary:

```
SRCDIR="$(dirname "${BASH_SOURCE[0]}")"
BASEDIR="$(dirname $(dirname $SRCDIR))"
XRHIDGEN="${BASEDIR}/tools/bin/xrhidgen"

if [[ ! -x $XRHIDGEN ]]; then
    error "${XRHIDGEN} is missing, run 'make install-tools'"
    exit 2
fi
```

tiran commented on test/scripts/ephe-domains-delete.sh:
```diff
@@ -1,24 +1,12 @@
 #!/bin/bash
```
Add `set -eo pipefail` to all shell scripts

```suggestion
#!/bin/bash
set -eo pipefail
```

avisiedo replied:
:+1:  thanks, I have added the change to the `common.inc` file.

avisiedo replied:
I have replicated this at the beginning of every script.

tiran commented on test/scripts/ephe-domains-delete.sh:
```diff
@@ -1,24 +1,12 @@
-}
-
-# make ephemeral-db-cli <<< "select domain_uuid from domains order by id desc limit 1;\\q"
+source "./test/scripts/ephe.inc.sh"
```
I recommend that you do not rely on current working dir for source includes. Instead use this construct for all source includes:

```suggestion
source "$(dirname "${BASH_SOURCE[0]}")/common.inc.sh"
```

avisiedo replied:
I like this! thanks!

tiran commented on test/scripts/ephe-domains-delete.sh:
```diff
@@ -1,24 +1,12 @@
-export X_RH_FAKE_IDENTITY="$( ./tools/bin/xrhidgen -org-id 12345 system -cn "6f324116-b3d2-11ed-8a37-482ae3863d30" -cert-type system | base64 -w0 )"
+export X_RH_FAKE_IDENTITY="${X_RH_FAKE_IDENTITY:-${IDENTITY_USER}}"
 export X_RH_IDM_REGISTRATION_TOKEN="${TOKEN}"
 export X_RH_IDM_VERSION="$( base64 -w0 <<< '{"ipa-hcc": "0.7", "ipa": "4.10.0-8.el9_1"}' )"
```
The IdM version variable is repeated a bunch of times. I suggest you add a function to `common.inc`
```
idm_version() {
    base64 -w0 <<< '{"ipa-hcc": "0.7", "ipa": "4.10.0-8.el9_1"}'
}
```

and replace all `X_RH_IDM_VERSION` with 

```suggestion
export X_RH_IDM_VERSION="$(idm_version)"
```

tiran commented on test/scripts/common.inc.sh:
```diff
@@ -0,0 +1,17 @@
+# Include file with common parts shared for local and ephemeral
+#
+
+function error {
```
Since it's a Bash script, not a POSIX sh script, use `()` to define functions:

```suggestion
error() {
```

avisiedo replied:
I don't see anything against that in the documentation at: https://www.gnu.org/software/bash/manual/bash.html#Shell-Functions ; if I am missing something happy to know.

tiran commented:
```
$ shellcheck -x -s bash -P test/scripts/ test/scripts/*

In test/scripts/common.inc line 26:
BASEDIR="$(dirname "$(dirname "${SRCDIR}")")"
^-----^ SC2034 (warning): BASEDIR appears unused. Verify use (or export if used externally).


In test/scripts/ephe-domains-delete.sh line 11:
export X_RH_IDM_VERSION="$( base64 -w0 <<< '{"ipa-hcc": "0.7", "ipa": "4.10.0-8.el9_1"}' )"
       ^--------------^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In test/scripts/ephe-domains-patch.sh line 13:
"${REPOBASEDIR}/scripts/curl.sh" -i -X PATCH -d @<( cat test/data/http/patch-rhel-idm-domain.json | sed -e "s/{{createDomain.response.body.domain_id}}/${UUID}/g" -e 's/{{subscription_manager_id}}/6f324116-b3d2-11ed-8a37-482ae3863d30/g' ) "${BASE_URL}/domains/${UUID}"
                                                        ^-- SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In test/scripts/ephe-domains-register.sh line 12:
export X_RH_IDM_VERSION="$(idm_version)"
       ^--------------^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In test/scripts/ephe-domains-register.sh line 13:
"${REPOBASEDIR}/scripts/curl.sh" -i -X POST -d @<( cat "${REPOBASEDIR}/test/data/http/register-rhel-idm-domain.json" | sed -e 's/{{subscription_manager_id}}/6f324116-b3d2-11ed-8a37-482ae3863d30/g' ) "${BASE_URL}/domains"
                                                       ^-- SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In test/scripts/ephe-domains-update.sh line 12:
export X_RH_IDM_VERSION="$(idm_version)"
       ^--------------^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In test/scripts/ephe-domains-update.sh line 13:
"${REPOBASEDIR}/scripts/curl.sh" -i -X PUT -d @<( cat test/data/http/update-rhel-idm-domain.json | sed -e "s/{{createDomain.response.body.domain_id}}/${UUID}/g" -e 's/{{subscription_manager_id}}/6f324116-b3d2-11ed-8a37-482ae3863d30/g' ) "${BASE_URL}/domains/${UUID}"
                                                      ^-- SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In test/scripts/ephe-hostconf.sh line 13:
export X_RH_IDM_VERSION="$(idm_version)"
       ^--------------^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In test/scripts/ephe.inc line 12:
export NAMESPACE="$(oc project -q)"
       ^-------^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In test/scripts/ephe.inc line 13:
CREDS="$( oc get secrets/env-${NAMESPACE}-keycloak -o jsonpath='{.data.defaultUsername}' | base64 -d )"
                             ^----------^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
CREDS="$( oc get secrets/env-"${NAMESPACE}"-keycloak -o jsonpath='{.data.defaultUsername}' | base64 -d )"


In test/scripts/ephe.inc line 14:
CREDS="${CREDS}:$( oc get secrets/env-${NAMESPACE}-keycloak -o jsonpath='{.data.defaultPassword}' | base64 -d )"
                                      ^----------^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
CREDS="${CREDS}:$( oc get secrets/env-"${NAMESPACE}"-keycloak -o jsonpath='{.data.defaultPassword}' | base64 -d )"


In test/scripts/ephe.inc line 17:
BASE_URL="https://$( oc get routes -l app=idmsvc-backend -o jsonpath='{.items[0].spec.host}' )/api/idmsvc/v1"
^------^ SC2034 (warning): BASE_URL appears unused. Verify use (or export if used externally).


In test/scripts/local-domains-patch.sh line 11:
"${REPOBASEDIR}/scripts/curl.sh" -i -X PATCH -d @<( cat "test/data/http/patch-rhel-idm-domain.json" | sed -e "s/{{createDomain.response.body.domain_id}}/${UUID}/g" ) "${BASE_URL}/domains/${UUID}"
                                                        ^-- SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In test/scripts/local-domains-populate.py line 14:
CONTENT_TYPE = "application/json"
             ^-- SC2283 (error): Remove spaces around = to assign (or use [ ] to compare, or quote '=' if literal).


In test/scripts/local-domains-populate.py line 16:
HEADER_CONTENT_TYPE = "Content-Type"
                    ^-- SC2283 (error): Remove spaces around = to assign (or use [ ] to compare, or quote '=' if literal).


In test/scripts/local-domains-populate.py line 17:
HEADER_X_RH_IDENTITY = "X-Rh-Identity"
                     ^-- SC2283 (error): Remove spaces around = to assign (or use [ ] to compare, or quote '=' if literal).


In test/scripts/local-domains-populate.py line 18:
HEADER_X_RH_INSIGHTS_REQUEST_ID = "X-Rh-Insights-Request-Id"
                                ^-- SC2283 (error): Remove spaces around = to assign (or use [ ] to compare, or quote '=' if literal).


In test/scripts/local-domains-populate.py line 19:
HEADER_X_RH_IDM_VERSION = "X-Rh-Idm-Version"
                        ^-- SC2283 (error): Remove spaces around = to assign (or use [ ] to compare, or quote '=' if literal).


In test/scripts/local-domains-populate.py line 20:
HEADER_X_RH_IDM_REGISTRATION_TOKEN = "X-Rh-Idm-Registration-Token"
                                   ^-- SC2283 (error): Remove spaces around = to assign (or use [ ] to compare, or quote '=' if literal).


In test/scripts/local-domains-populate.py line 22:
DEFAULT_ORG_ID = os.environ.get("ORG_ID", "12345")
               ^-- SC2283 (error): Remove spaces around = to assign (or use [ ] to compare, or quote '=' if literal).
                               ^-- SC1036 (error): '(' is invalid here. Did you forget to escape it?
                               ^-- SC1088 (error): Parsing stopped here. Invalid use of parentheses?


In test/scripts/local-domains-register.sh line 11:
export X_RH_IDM_VERSION="$(idm_version)"
       ^--------------^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In test/scripts/local-domains-register.sh line 12:
"${REPOBASEDIR}/scripts/curl.sh" -i -X POST -d @<( cat "${REPOBASEDIR}/test/data/http/register-rhel-idm-domain.json" | sed -e 's/{{subscription_manager_id}}/6f324116-b3d2-11ed-8a37-482ae3863d30/g' ) "${BASE_URL}/domains"
                                                       ^-- SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In test/scripts/local-domains-update.sh line 11:
export X_RH_IDM_VERSION="$(idm_version)"
       ^--------------^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In test/scripts/local-domains-update.sh line 12:
"${REPOBASEDIR}/scripts/curl.sh" -i -X PUT -d @<( cat "${REPOBASEDIR}/test/data/http/update-rhel-idm-domain.json" | sed -e 's/{{subscription_manager_id}}/6f324116-b3d2-11ed-8a37-482ae3863d30/g' ) "${BASE_URL}/domains/${UUID}"
                                                      ^-- SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In test/scripts/local-hostconf.sh line 11:
export X_RH_IDM_VERSION="$(idm_version)"
       ^--------------^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In test/scripts/local.inc line 6:
BASE_URL="http://localhost:8000/api/idmsvc/v1"
^------^ SC2034 (warning): BASE_URL appears unused. Verify use (or export if used externally).

For more information:
  https://www.shellcheck.net/wiki/SC2283 -- Remove spaces around = to assign ...
  https://www.shellcheck.net/wiki/SC2034 -- BASEDIR appears unused. Verify us...
  https://www.shellcheck.net/wiki/SC2155 -- Declare and assign separately to ...
[heimes@seneca idm-domains-backend]$ shellcheck -x -s bash -P test/scripts/ test/scripts/*.sh test/scripts/*.inc

In test/scripts/ephe-domains-delete.sh line 11:
export X_RH_IDM_VERSION="$( base64 -w0 <<< '{"ipa-hcc": "0.7", "ipa": "4.10.0-8.el9_1"}' )"
       ^--------------^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In test/scripts/ephe-domains-patch.sh line 13:
"${REPOBASEDIR}/scripts/curl.sh" -i -X PATCH -d @<( cat test/data/http/patch-rhel-idm-domain.json | sed -e "s/{{createDomain.response.body.domain_id}}/${UUID}/g" -e 's/{{subscription_manager_id}}/6f324116-b3d2-11ed-8a37-482ae3863d30/g' ) "${BASE_URL}/domains/${UUID}"
                                                        ^-- SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In test/scripts/ephe-domains-register.sh line 12:
export X_RH_IDM_VERSION="$(idm_version)"
       ^--------------^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In test/scripts/ephe-domains-register.sh line 13:
"${REPOBASEDIR}/scripts/curl.sh" -i -X POST -d @<( cat "${REPOBASEDIR}/test/data/http/register-rhel-idm-domain.json" | sed -e 's/{{subscription_manager_id}}/6f324116-b3d2-11ed-8a37-482ae3863d30/g' ) "${BASE_URL}/domains"
                                                       ^-- SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In test/scripts/ephe-domains-update.sh line 12:
export X_RH_IDM_VERSION="$(idm_version)"
       ^--------------^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In test/scripts/ephe-domains-update.sh line 13:
"${REPOBASEDIR}/scripts/curl.sh" -i -X PUT -d @<( cat test/data/http/update-rhel-idm-domain.json | sed -e "s/{{createDomain.response.body.domain_id}}/${UUID}/g" -e 's/{{subscription_manager_id}}/6f324116-b3d2-11ed-8a37-482ae3863d30/g' ) "${BASE_URL}/domains/${UUID}"
                                                      ^-- SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In test/scripts/ephe-hostconf.sh line 13:
export X_RH_IDM_VERSION="$(idm_version)"
       ^--------------^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In test/scripts/local-domains-patch.sh line 11:
"${REPOBASEDIR}/scripts/curl.sh" -i -X PATCH -d @<( cat "test/data/http/patch-rhel-idm-domain.json" | sed -e "s/{{createDomain.response.body.domain_id}}/${UUID}/g" ) "${BASE_URL}/domains/${UUID}"
                                                        ^-- SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In test/scripts/local-domains-register.sh line 11:
export X_RH_IDM_VERSION="$(idm_version)"
       ^--------------^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In test/scripts/local-domains-register.sh line 12:
"${REPOBASEDIR}/scripts/curl.sh" -i -X POST -d @<( cat "${REPOBASEDIR}/test/data/http/register-rhel-idm-domain.json" | sed -e 's/{{subscription_manager_id}}/6f324116-b3d2-11ed-8a37-482ae3863d30/g' ) "${BASE_URL}/domains"
                                                       ^-- SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In test/scripts/local-domains-update.sh line 11:
export X_RH_IDM_VERSION="$(idm_version)"
       ^--------------^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In test/scripts/local-domains-update.sh line 12:
"${REPOBASEDIR}/scripts/curl.sh" -i -X PUT -d @<( cat "${REPOBASEDIR}/test/data/http/update-rhel-idm-domain.json" | sed -e 's/{{subscription_manager_id}}/6f324116-b3d2-11ed-8a37-482ae3863d30/g' ) "${BASE_URL}/domains/${UUID}"
                                                      ^-- SC2002 (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.


In test/scripts/local-hostconf.sh line 11:
export X_RH_IDM_VERSION="$(idm_version)"
       ^--------------^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In test/scripts/common.inc line 26:
BASEDIR="$(dirname "$(dirname "${SRCDIR}")")"
^-----^ SC2034 (warning): BASEDIR appears unused. Verify use (or export if used externally).


In test/scripts/ephe.inc line 12:
export NAMESPACE="$(oc project -q)"
       ^-------^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In test/scripts/ephe.inc line 13:
CREDS="$( oc get secrets/env-${NAMESPACE}-keycloak -o jsonpath='{.data.defaultUsername}' | base64 -d )"
                             ^----------^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
CREDS="$( oc get secrets/env-"${NAMESPACE}"-keycloak -o jsonpath='{.data.defaultUsername}' | base64 -d )"


In test/scripts/ephe.inc line 14:
CREDS="${CREDS}:$( oc get secrets/env-${NAMESPACE}-keycloak -o jsonpath='{.data.defaultPassword}' | base64 -d )"
                                      ^----------^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
CREDS="${CREDS}:$( oc get secrets/env-"${NAMESPACE}"-keycloak -o jsonpath='{.data.defaultPassword}' | base64 -d )"


In test/scripts/ephe.inc line 17:
BASE_URL="https://$( oc get routes -l app=idmsvc-backend -o jsonpath='{.items[0].spec.host}' )/api/idmsvc/v1"
^------^ SC2034 (warning): BASE_URL appears unused. Verify use (or export if used externally).


In test/scripts/local.inc line 6:
BASE_URL="http://localhost:8000/api/idmsvc/v1"
^------^ SC2034 (warning): BASE_URL appears unused. Verify use (or export if used externally).

For more information:
  https://www.shellcheck.net/wiki/SC2034 -- BASEDIR appears unused. Verify us...
  https://www.shellcheck.net/wiki/SC2155 -- Declare and assign separately to ...
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
```

avisiedo commented:
@tiran thanks for the review; I have updated the linter violations; the PR is ready for another review.

tiran commented on test/scripts/common.inc:
```diff
@@ -0,0 +1,63 @@
+# - Run with more verbose by:
+#   $ DEBUG=1 bash -xv ./test/scripts/local-domain-token.sh
+
+set -eo pipefail
```
Shell settings are typically set in the beginning of a shell script, not in an include source file. It allows a developer to source an include into an interactive session without setting `-e`.

tiran commented on test/scripts/common.inc:
```diff
@@ -0,0 +1,63 @@
+export IDM_VERSION=""     # Use $(idm_version)
+
+SRCDIR="$(dirname "${BASH_SOURCE[0]}")"
+# shellcheck disable=SC2034
```
```suggestion
# shellcheck disable=SC2034  # ignore unused variable
```

tiran commented on test/scripts/common.inc:
```diff
@@ -0,0 +1,63 @@
+    exit 2
+fi
+
+function identity_user() {
+    if [ "${IDENTITY_USER}" == "" ]; then
+        IDENTITY_USER="$( "${XRHIDGEN}" -org-id "${ORG_ID}" user -is-active=true -is-org-admin=true -user-id test -username test | base64 -w0 )"
+        [ "${IDENTITY_USER}" != "" ] || error "IDENTITY_USER is empty"
+    fi
+    printf "%s" "${IDENTITY_USER}"
+}
```
The function looks unnecessarily complicated. Why do you cache the value? What is wrong with:

```suggestion
function identity_user() {
    "${XRHIDGEN}" -org-id "${ORG_ID}" user -is-active=true -is-org-admin=true -user-id test -username test | base64 -w0}
```

avisiedo replied:
nothing is wrong, I thought it could be better! but rethinking, if we are calling at some point with a different place with other ORG_ID, the returned output would be wrong; so yes, better do not cache and reduce complexity.

tiran commented on test/scripts/ephe-domains-delete.sh:
```diff
@@ -1,24 +1,13 @@
-export X_RH_IDM_VERSION="$( base64 -w0 <<< '{"ipa-hcc": "0.7", "ipa": "4.10.0-8.el9_1"}' )"
-BASE_URL="https://$( oc get routes -l app=idmsvc-backend -o jsonpath='{.items[0].spec.host}' )/api/idmsvc/v1"
-./scripts/curl.sh -i -X DELETE "${BASE_URL}/domains/${UUID}"
+X_RH_IDM_VERSION="$( base64 -w0 <<< '{"ipa-hcc": "0.7", "ipa": "4.10.0-8.el9_1"}' )"
```
Use `idm_version` here.

tiran commented on test/scripts/ephe-domains-patch.sh:
```diff
@@ -1,24 +1,13 @@
-export X_RH_FAKE_IDENTITY="$( ./bin/xrhidgen -org-id 12345 system -cn "6f324116-b3d2-11ed-8a37-482ae3863d30" -cert-type system | base64 -w0 )"
+export X_RH_FAKE_IDENTITY="${X_RH_FAKE_IDENTITY:-$(identity_user)}"
 unset X_RH_IDM_REGISTRATION_TOKEN
 export X_RH_IDM_VERSION='{"ipa-hcc": "0.9", "ipa": "4.10.0-8.el9_1", "os-release-id": "rhel", "os-release-version-id": "9.1"}'
```
Use `idm_version` here

tiran commented on test/scripts/ephe.inc:
```diff
@@ -0,0 +1,19 @@
+
+NAMESPACE="$(oc project -q)"
+export NAMESPACE
+CREDS="$( oc get secrets/env-"${NAMESPACE}"-keycloak -o jsonpath='{.data.defaultUsername}' | base64 -d )"
+CREDS="${CREDS}:$( oc get secrets/env-"${NAMESPACE}"-keycloak -o jsonpath='{.data.defaultPassword}' | base64 -d )"
+export CREDS
```
```suggestion
username="$( oc get secrets/env-"${NAMESPACE}"-keycloak -o jsonpath='{.data.defaultUsername}' | base64 -d )"
password="$( oc get secrets/env-"${NAMESPACE}"-keycloak -o jsonpath='{.data.defaultPassword}' | base64 -d )"
CREDS="${username}:${password}"
export CREDS
```

tiran reviewed (changes requested):
`shellcheck disable=SC1091` is not necessary with `shellcheck -x` and correct `-P` include path: 
> ```shellcheck -x -s bash -P test/scripts/ test/scripts/*.sh test/scripts/*.inc```

`set -eo pipefail` is one of the few cases, where "repeat yourself" is better than DRY. I suggest that you move them into the header of each shell script.

tiran committed:
fix: Shell script improvements

- add `make shellcheck`
- Removed unused IDENTITY_USER and IDENTITY_SYSTEM
- Declare IDM_VERSION with static content
- export is not necessary for source files
- bash style: `function` prefix is not necessary
- Drop unnecessary `shellcheck disable=SC1091`
- exec `curl.sh`

Signed-off-by: Christian Heimes <cheimes@redhat.com>

tiran commented:
@avisiedo I have added a commit on top of your changes with some style improvements and fixes.

avisiedo commented:
I am ok with the last changes; @tiran  any further concerns???
